presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -149,32 +164,17 @@ public Principal getUserPrincipal()
         return environment;
     }
 
-    private String getPrincipal(String user, DirContext ctx)
+    /**
+     * Returns the prinicipal which will be used to bind to LDAP
+     *
+     * @param user
+     * @return
+     * @throws NamingException
+     */
+    protected String getPrincipal(String user)
```

Comment:
I would rather keep this method abstract instead of providing default implementation which is never used. But you can leave if you think that it does make sense.